app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java

```diff
@@ -299,7 +300,15 @@ private View buildCatLabel(final String catName, ViewGroup categoryContainer) {
                 Intent viewIntent = new Intent();
                 viewIntent.setAction(Intent.ACTION_VIEW);
                 viewIntent.setData(new PageTitle(selectedCategoryTitle).getCanonicalUri());
-                startActivity(viewIntent);
+                //check if web browser available
+                if(viewIntent.resolveActivity(getActivity().getPackageManager()) != null){
```

Comment:
Instead of duplicating the same logic to all the activity files, extract out this logic to some other util class ie deeplink resolver or something similar. :)

Reply:
I think this might be least verbose method to do this. But I would love more info on deep-linking as I am not aware of this :)

Reply:
@maskaravivek Any updates? :)

Reply:
I was talking about extracting the whole logic to a util method like

```
Util.handleWebUrl(context, url);
```

Its just a suggested refactoring. I dont think its a blocker for merging the PR. :)

Reply:
Thanks for the knowledge! :)